migrations/helpers/update_filtered_positives_helper.py

```diff
@@ -0,0 +1,270 @@
+
+        # this should take everything from the cursor find into RAM memory
+        # (assuming you have enough memory)
+        # should we project to an object that has fewer fields?
+        return list(
+            samples_collection.find(
+                {
+                    FIELD_RESULT: {"$eq": POSITIVE_RESULT_VALUE},
+                    FIELD_PLATE_BARCODE: {"$in": plate_barcodes},
+                }
+            )
+        )
+
+
+def update_filtered_positive_fields(
+    filtered_positive_identifier: FilteredPositiveIdentifier,
+    samples: List[Dict[str, Any]],
+    version: str,
+    update_timestamp: datetime,
+) -> None:
+    """Updates filtered positive fields on all passed-in samples
+
+    Arguments:
+        filtered_positive_identifier {FilteredPositiveIdentifier} -- the identifier through which
+        to pass samples to determine whether they are filtered positive
+
+        samples {List[Dict[str, str]]} -- the list of samples for which to re-determine filtered
+        positive values
+
+        version {str} -- the filtered positive identifier version used
+        update_timestamp {datetime} -- the timestamp at which the update was performed
+    """
+    version = filtered_positive_identifier.current_version()
+    # Expect all samples to be passed into here to have a positive result
+    for sample in samples:
+        sample[FIELD_FILTERED_POSITIVE] = filtered_positive_identifier.is_positive(sample)
+        sample[FIELD_FILTERED_POSITIVE_VERSION] = version
+        sample[FIELD_FILTERED_POSITIVE_TIMESTAMP] = update_timestamp
+
+
+def update_mongo_filtered_positive_fields(
+    config: ModuleType, samples: List[Dict[str, Any]], version: str, update_timestamp: datetime
+) -> bool:
+    """Bulk updates sample filtered positive fields in the Mongo database
+
+    Arguments:
+        config {ModuleType} -- application config specifying database details
+        samples {List[Dict[str, str]]} -- the list of samples whose filtered positive fields
+        should be updated
+
+        version {str} -- the filtered positive identifier version used
+        update_timestamp {datetime} -- the timestamp at which the update was performed
+
+    Returns:
+        bool -- whether the updates completed successfully
+    """
+    with create_mongo_client(config) as client:
+        mongo_db = get_mongo_db(config, client)
+
+        # get ids of those that are filtered positive, and those that aren't
+        all_ids = [sample[FIELD_MONGODB_ID] for sample in samples]
+        filtered_positive_ids = [
+            sample[FIELD_MONGODB_ID]
+            for sample in list(filter(lambda x: x[FIELD_FILTERED_POSITIVE] is True, samples))
+        ]
+        filtered_negative_ids = [
+            mongo_id for mongo_id in all_ids if mongo_id not in filtered_positive_ids
+        ]
+
+        samples_collection = get_mongo_collection(mongo_db, COLLECTION_SAMPLES)
+        samples_collection.update_many(
```

Comment:
Do these updates throw anything if something happens?

Reply:
Quite possibly, not sure on details. Any exceptions will get caught in the calling method